src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -136,17 +144,29 @@ void set_decoder_input_ids_attention_mask(ov::InferRequest& decoder,
     std::fill(attention_mask_data + init_ids.size(), attention_mask_data + attention_mask_tensor.get_size(), 0u);
 }
 
+void infer_with_perf_metrics(ov::InferRequest& request, ov::genai::RawPerfMetrics& raw_metrics) {
```

Comment:
Duplication from: https://github.com/openvinotoolkit/openvino.genai/blob/d91123a375c62a0012e3239adfb1c6fc14b4bcbf/src/cpp/src/whisper/whisper.cpp#L81-L90

@as-suvorov  does it make sense to re-use this function in both pipelines?

Reply:
Added also functions for non segment tokens filtering, which are also duplicate functions in whisper.cpp.
Maybe we could move common functions to common space and reuse them? (e.g. create whisper_utils.hpp/cpp for this)